src/peer_digest.cc

```diff
@@ -676,27 +659,26 @@ peerDigestFetchAbort(DigestFetchState * fetch, char *buf, const char *reason)
 {
     assert(reason);
     debugs(72, 2, "peerDigestFetchAbort: peer " << fetch->pd->host << ", reason: " << reason);
-    peerDigestReqFinish(fetch, buf, 1, 1, 1, reason, 1);
+    peerDigestReqFinish(fetch, buf, reason, 1);
 }
 
 /* complete the digest transfer, update stats, unlock/release everything */
+
 static void
-peerDigestReqFinish(DigestFetchState * fetch, char * /* buf */,
-                    int fcb_valid, int pdcb_valid, int pcb_valid,
-                    const char *reason, int err)
+peerDigestReqFinish(DigestFetchState * const fetch, char *, const char * const reason, const int err)
 {
     assert(reason);
 
-    /* must go before peerDigestPDFinish */
+    /* must go before PeerDigest::noteFetchFinished() */
```

Comment:
Why does the code below has to run before we call PeerDigest::noteFetchFinished()?